### PR TITLE
Feature/subscription

### PR DESCRIPTION
**Pull request by LeHaiDang67**

### CODE REVIEW

File: cmd/controller/controller.go [hunk 1]
```diff
@@ -107,3 +107,20 @@ func CommonFriends(db *sql.DB) http.HandlerFunc {
...
 }
+
+//Subscription is...
+func Subscription(db *sql.DB) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var subRequest model.SubscriptionRequest
+		err := json.NewDecoder(r.Body).Decode(&subRequest)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte("HTTP status code returned!"))
```
Comment (tanphat195) [inline cmd/controller/controller.go:118]: Should be `Request body is invalid`

File: cmd/controller/controller.go [hunk 2]
```diff
@@ -107,3 +107,20 @@ func CommonFriends(db *sql.DB) http.HandlerFunc {
...
+			return
+		}
+		result := repo.Subscription(db, subRequest)
+		json.NewEncoder(w).Encode(result)
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("HTTP status code returned!"))
```
Comment (tanphat195) [inline cmd/controller/controller.go:130]: Should be `Successfully subscript`

Comment (tanphat195) [reply cmd/controller/controller.go:130]: Not use json.NewEncoder(w).Encode(result) & w.Write([]byte("HTTP status code returned!")) together

File: intenal/feature/repo/repo.go [hunk 3]
```diff
@@ -160,6 +172,19 @@ func UpdateUser2(db *sql.DB, user FakeUser, email string) error {
 	return nil
 }
 
+//UpdateUser3 append the user []
+func UpdateUser3(db *sql.DB, requestor string, target string) error {
```
Comment (tanphat195) [inline intenal/feature/repo/repo.go:186]: Why don't use UpdateUser?

**Comment by tanphat195** [comment]
LGTM! There are some comments

File: intenal/feature/repo/repo.go (hunk 3 above)
Comment (LeHaiDang67) [reply intenal/feature/repo/repo.go:186]: UpdateUser 2 for friend
UpdateUser 3 for subscribe
UpdateUser 4 for blocked

**Pull request closed without merging**